activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -81,6 +81,31 @@ def visit_DropForeignKey(name)
           "DROP CONSTRAINT #{quote_column_name(name)}"
         end
 
+        def visit_CreateIndexDefinition(o)
+          index = o.index
+
+          sql = ["CREATE"]
+          sql << "UNIQUE" if index.unique
+          sql << "INDEX"
+          sql << "IF NOT EXISTS" if o.if_not_exists
+          sql << o.algorithm if o.algorithm
+          sql << index.type if index.type
+          sql << "#{quote_column_name(index.name)} ON #{quote_table_name(index.table)}"
+          sql << "USING #{index.using}" if supports_index_using? && index.using
```

Comment:
Is this a behavior change? Looks like `support_index_using?` [was not checked previously](https://github.com/rails/rails/pull/39203/files#diff-a0775e1ec64264dc76a8ee71a5211ae3L1238).

Reply:
Nope, This is to keep the behavior as before.

Before, the code generation is not shared, now it is shared by SQLite3 and PostgreSQL, but SQLite3 doesn't support USING syntax, so it will raise syntax error unless the `supports_index_using?` guard.

Reply:
SQLite3:

https://github.com/rails/rails/blob/1c84b1ab50b9c562cdd407be5827875982de07aa/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L823

PostgreSQL:

https://github.com/rails/rails/blob/1c84b1ab50b9c562cdd407be5827875982de07aa/activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb#L446